From 7f578bb42c4711682e304e434634bfdad053493b Mon Sep 17 00:00:00 2001
From: Amelia Crate <amelia.crate@chainguard.dev>
Date: Tue, 24 Jun 2025 08:46:21 -0700
Subject: [PATCH 1/4] crypto: reusable jitterentropy

log build-time kernel version in jent init, so that it is identifiable at runtime
---
 crypto/jitterentropy.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/crypto/jitterentropy.c b/crypto/jitterentropy.c
index 3b390bd6c1196..0bdbaeb40bbbd 100644
--- a/crypto/jitterentropy.c
+++ b/crypto/jitterentropy.c
@@ -147,6 +147,8 @@ struct rand_data {
 
 #include <linux/fips.h>
 #include <linux/minmax.h>
+#include <linux/printk.h>
+#include <generated/utsrelease.h>
 #include "jitterentropy.h"
 
 /***************************************************************************
@@ -707,6 +709,7 @@ void jent_entropy_collector_free(struct rand_data *entropy_collector)
 int jent_entropy_init(unsigned int osr, unsigned int flags, void *hash_state,
 		      struct rand_data *p_ec)
 {
+	printk(KERN_INFO "Initializing jitterentropy built from %s sources\n", UTS_RELEASE);
 	/*
 	 * If caller provides an allocated ec, reuse it which implies that the
 	 * health test entropy data is used to further still the available
-- 
2.50.1

